actions/treasures.go

```diff
@@ -15,7 +24,51 @@ type treasureReq struct {
 	EthKey          string `json:"EthKey"`
 }
 
+type treasureRes struct {
+	Success string `json:"success"`
+}
+
+func unixMilli(t time.Time) int64 {
+	return t.Round(time.Millisecond).UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
-	return nil
-}
+	req := treasureReq{}
+	oyster_utils.ParseReqBody(c.Request(), &req)
+
+	host_ip := os.Getenv("HOST_IP")
+	provider := "http://" + host_ip + ":14265"
+	api := giota.NewAPI(provider, nil)
+	ftr := &giota.FindTransactionsRequest{Bundles: []giota.Trytes{"DEXRPLKGBROUQMKCLMRPG9HFKCACDZ9AB9HOJQWERTYWERJNOYLW9PKLOGDUPC9DLGSUH9UHSKJOASJRU"}}
+	resp, err := api.FindTransactions(ftr)
+
+	datamap1, vErr := models.GetDataMap(req.GenesisHash, req.NumChunks)
+	for _, d := range datamap1 {
+		
+	}
+	// or 
+	
+	vErr, error := models.BuildDataMaps(req.GenesisHash, req.NumChunks)
+	datamap, err := models.GetDataMapByGenesisHashAndChunkIdx(req.GenesisHash, req.NumChunks)
+	for _, d := range datamap {
+		
+	}
+	
+	var transactions[2] int64
+	transactions[0] = 11
+	transactions[0] = 1122
```

Comment:
What are these for?  It looks like you're setting transactions[0] then immediately setting it to something else?

Reply:
Its dummy transaction

Reply:
Gotcha